web/client/components/mapcontrols/annotations/SelectAnnotationsFile.jsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2017, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const React = require('react');
+const PropTypes = require('prop-types');
+
+const Dropzone = require('react-dropzone');
+const Spinner = require('react-spinkit');
+const {Checkbox} = require("react-bootstrap");
+
+const {isEmpty} = require("lodash");
+
+const Message = require('../../I18N/Message');
+const ResizableModal = require('../../misc/ResizableModal');
+
+const LocaleUtils = require('../../../utils/LocaleUtils');
+
+const FileUtils = require('../../../utils/FileUtils');
+const {Promise} = require('es6-promise');
+
+class SelectAnnotationsFile extends React.Component {
+    static propTypes = {
+        show: PropTypes.bool,
+        text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+        onFileChoosen: PropTypes.func,
+        error: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+        errorMessage: PropTypes.string,
+        onClose: React.PropTypes.func,
+        title: PropTypes.node,
+        closeGlyph: PropTypes.string,
+        diableOvveride: React.PropTypes.bool
+    };
+
+    static contextTypes = {
+        messages: PropTypes.object
+    };
+
+    static defaultProps = {
+        show: false,
+        text: "Drop your file here or click to select the Annotation File to import. (supported files: JSON)",
```

Comment:
is this not localized?

Reply:
Yes it is
The message is passed from the container componet